swarm/storage/pyramid.go

```diff
@@ -75,15 +75,61 @@ const (
 	TreeChunk = 1
 )
 
-type ChunkerParams struct {
-	Branches int64
-	Hash     string
+type PyramidSplitterParams struct {
+	SplitterParams
+	getter Getter
 }
 
-func NewChunkerParams() *ChunkerParams {
-	return &ChunkerParams{
-		Branches: DefaultBranches,
-		Hash:     SHA3Hash,
+func NewPyramidSplitterParams(key Key, reader io.Reader, putter Putter, getter Getter, branches int64) *PyramidSplitterParams {
+	hashSize := putter.RefSize()
+	return &PyramidSplitterParams{
+		SplitterParams: SplitterParams{
+			ChunkerParams: ChunkerParams{
+				// TODO: use chunkSize as parameter, because that is the constant
+				branches: 4096 / hashSize,
+				hashSize: hashSize,
+			},
+			reader: reader,
+			putter: putter,
+			key:    key,
+		},
+		getter: getter,
+	}
+}
+
+func NewTreeSplitterParams(reader io.Reader, putter Putter, size int64, branches int64) *TreeSplitterParams {
```

Comment:
Why is this in `pyramid.go` when `TreeSplitter` type declaration is in `chunker.go`?